### ISSUE TITLE
don't show add-ons that don't work on my platform!

### ISSUE THREAD

**Issue by unbelievable-u-missed** [mozilla/addons#667]
### Describe the problem and steps to reproduce it:
(Please include as many details as possible.)
I look for AddOns. The URL includes, "android", so why are you listing things that don't work in this platform? But I don't know until I click and there it is, the red bar.
### What did you expect to happen?
Don't list non android, or non whatever, in the page.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)
It's hard to think of what else you need to know, when to me this is an example of something you should have already known. 

You guys know about taking small bites and chewing your food completely before swallowing. And if you forget and choke, lift your hands above your head while you cough? I hope so.

**Comment by tofumatt** [comment]
Your hostile tone does not help solve your issue nor does it shed light on the search you performed or the add-on(s) that caused the error. We're here to help, and this error is probably an honest mistake or a bug. These things happen, but when reported in a friendly (or even understandably frustrated) tone: they get taken seriously. When they're framing in a mocking tone, it's tough to empathise with the issue and it's taken less seriously, at least for me.

You omitted the URL and exact search that caused the issue, but had time to type out a sarcastic insult  to the developers of the site you're using and asking to make improvements on. I understand you're frustrated, but I hope next time you file an issue with a project you remember there are hard-working people on the other end of the comment and that we want to help.

That being said, we still want to help: could you include the add-on you searched for and, if possible, the exact URL causing the error? It would also help to know your platform (I'm guessing Android, but just want to make sure you're searching from mobile 😄) and your browser version number.

**Comment by unbelievable-u-missed** [comment]
https://addons.mozilla.org/en-US/android/
then I scrolled down and picked "Update Scanner"
![capture1](https://user-images.githubusercontent.com/37462117/50726148-39c08c80-10c6-11e9-8075-e3f1dc413aad.JPG)

and the red line shows up.
![capture](https://user-images.githubusercontent.com/37462117/50726127-e6e6d500-10c5-11e9-8545-3719b09aa222.JPG)

**Comment by unbelievable-u-missed** [comment]
So, if I'm looking for things for android, why is it listing things that won't work? That's what my original post is about.

**Comment by smayer97** [comment]
That's a tough call... what if you are looking at addons with another browser, then NOTHING would show up  :-\

**Comment by unbelievable-u-missed** [comment]
No, you're not getting it. If I specifically chose Android on the previous
page, then that filter should be applied as the act of choosing would and
should imply.

I have no idea what you're talking about. But to follow it to an illogical
end, suppose the setting was language. You pick EN "und dann ist die
Webseite unklar?" That's not a user issue. This is a problem with the
site's code.

Does this yet more broken down example help?

Thanks

**Comment by jvillalobos** [comment]
This is a valid issue, but it's known and it's not something we consider worth fixing right now. The homepage is comprised of curated content, including collections, which is selected with desktop Firefox in mind. Most content on the site works both on desktop and mobile, but there are extensions like Update Scanner that only work on desktop.

Filtering them out has been considered, but it's more effort than it warrants for a relatively small user population.
